andru1989/vowels.rb

```diff
@@ -37,26 +40,57 @@ def read_file
     print_vowels
   end
 
+  # Public
+  #
+  #   => Set initial accessor with Time.now
+  #   => Get the content of 'text.txt' file and loop each slice with(20000)
+  #   => Acumulate in '@threads' Array a new Thread
+  #   => For each ary it call the 'set_values' method to update de Hash
+  #   => Wait a moment while each Thread finish
+  #   => Set final accessor with Time.now
+  #   => Then it call the 'print_vowels' method to print the hash
+  def read_thread_file
+    self.initial = Time.now
+    open_file.each_slice(20000) do |ary|
+      @threads << Thread.new do
+        set_values(ary)
+      end
+    end
+    @threads.each{|t| t.join}
+    self.final = Time.now
+
+    print_vowels
+  end
+
   private
   # Private:
   #
   #   => Print the Hash @vowels
   #   => Print the difference between initial and final accessors to show how many seconds the process long
+  #   => Set 0 for each key of the hash
   def print_vowels
     p @vowels
     p "Run in #{(self.final - self.initial)} Seconds"
+    @vowels.keys.each{|k| k = 0}
```

Comment:
Unused block argument - `k`. You can omit the argument if you don't care about it.<br>Useless assignment to variable - `k`.<br>Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.